uamqp/client.py

```diff
@@ -260,7 +260,11 @@ def open(self, connection=None):
             self._build_session()
             if self._keep_alive_interval:
                 self._keep_alive_thread = threading.Thread(target=self._keep_alive)
-                self._keep_alive_thread.start()
+                try:
+                    self._keep_alive_thread.start()
+                except RuntimeError:
+                    self._keep_alive_thread = None
```

Comment:
I personally feel it's simpler to add try/except around `join` in the `close` method instead of mutating the variable on the client during the open stage.

also, through this approach, we could handle other thread-related runtime error unlimited to thread not started.

```python
    def close(self):
        if self._keep_alive_thread:
            try:
                self._keep_alive_thread.join()
            except RuntimeError as exc:
                logging.info("keep alive thread failed to join due to %r", exc)
            finally:
                self._keep_alive_thread = None
```

Reply:
Fine with me.
Not sure I see any reason for the `finally`, so have skipped that line. Please check changes now.